### PR TITLE
Add docker validation jobs for COS

### PR DESCRIPTION
**Pull request by kewu1992**
@wonderfly Please review.

### CODE REVIEW

File: jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml [hunk 1]
```diff
@@ -717,6 +717,12 @@
         jenkins-timeout: 170
         timeout: 70
         trigger-job: 'ci-kubernetes-build'
+    - kubernetes-e2e-gce-gci-qa-docker13:
```
Comment (wonderfly) [inline jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml]: Suggest s/gce-gci-qa-docker13/cos-docker-validation/
For 1, qa doesn't make sense here, and 2, we might want to reuse the set of jobs for other docker versions so lets leave the name version-neutral.

File: jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml [hunk 2]
```diff
@@ -717,6 +717,12 @@
...
+        frequency: 'H/5 * * * *' # At least every 30m
+        job-name: ci-kubernetes-e2e-gce-gci-qa-docker13
+        jenkins-timeout: 170
+        timeout: 70
+        trigger-job: 'ci-kubernetes-build'
```
Comment (wonderfly) [inline jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml]: I don't think we need a trigger job. Just freq should be fine.

File: testgrid/config/config.yaml [hunk 3]
```diff
@@ -174,6 +174,8 @@ test_groups:
   gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-gci-ci-slow-release-1.5
 - name: ci-kubernetes-e2e-gci-gce-examples
   gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-examples
+- name: ci-kubernetes-e2e-gce-gci-qa-docker13
```
Comment (wonderfly) [inline testgrid/config/config.yaml]: Consider adding these to the `google-docker` tab, which was intended for docker validation tests

**Comment by kewu1992** [comment]
Done. Also add @Random-Liu for review.

File: jenkins/job-configs/kubernetes-jenkins/bootstrap-ci-repo.yaml [hunk 4]
```diff
@@ -194,6 +194,12 @@
         job-name: ci-kubernetes-node-docker-benchmark
         repo-name: k8s.io/kubernetes
         timeout: 90
+    - kubernetes-node-cos-docker-validation:  
+        branch: master
+        frequency: '@hourly'
+        job-name: ci-kubernetes-node-cos-docker-validation
```
Comment (wonderfly) [inline jenkins/job-configs/kubernetes-jenkins/bootstrap-ci-repo.yaml]: Is the `ci-kubernetes` prefix required? If not I think we can drop it

File: jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml [hunk 5]
```diff
@@ -717,6 +717,12 @@
         jenkins-timeout: 170
         timeout: 70
         trigger-job: 'ci-kubernetes-build'
+    - kubernetes-e2e-cos-docker-validation:
+        frequency: 'H/5 * * * *' # At least every 30m
```
Comment (wonderfly) [inline jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml]: Note that this is going to run the same tests with the same image over and over again. Once every 30 min is fine. We just need to remember to stop it once we have got enough passes.

**Comment by Random-Liu** [comment]
@kewu1992 LGTM. PR needs rebase

File: jenkins/job-configs/kubernetes-jenkins/bootstrap-ci-repo.yaml (hunk 4 above)
Comment (kewu1992) [inline jenkins/job-configs/kubernetes-jenkins/bootstrap-ci-repo.yaml]: I don't see any doc saying that it is required. But almost all jobs start with `ci-kubernetes` (others start with `pull-kubernetes` and `periodic-kubernetes`). So I think it is an implicit agreement for all CI test jobs.

**Comment by Random-Liu** [comment]
@k8s-bot ok to test

**Comment by wonderfly** [comment]
From the failed Bazel test:

```
AssertionError: ['ci-kubernetes-e2e-cos-docker-validation.env']: Env 'KUBE_GCE_NODE_PROJECT="${KUBE_GCE_MASTER_PROJECT}"': No quote in env
```
Seems relevant.

**Comment by wonderfly** [comment]
/lgtm

Note that the failing test can be verified locally, https://github.com/kubernetes/test-infra#building-and-testing-the-test-infra

File: jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml (hunk 5 above)
Comment (kewu1992) [inline jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml]: Sure. We could stop this as long as cos-dev-60 picked up docker 13.

**Comment by kewu1992** [comment]
I also manually ran the tests and verified that the *.env files are correct (test jobs could tear up and tear down correctly, although some test cases are failed).

**Pull request merged**